spec.html

```diff
@@ -42226,6 +42406,123 @@ <h1>Set.prototype.has ( _value_ )</h1>
+                  1. If _alreadyInResult_ is *false*, then
+                    1. Append _e_ to _resultSetData_.
+                1. NOTE: The number of elements in _O_.[[SetData]] may have increased during execution of _otherRec_.[[Has]].
+                1. Set _thisSize_ to the number of elements in _O_.[[SetData]].
+          1. Else,
+            1. Let _keysIter_ be ? GetIteratorFromMethod(_otherRec_.[[Set]], _otherRec_.[[Keys]]).
+            1. Let _next_ be *true*.
+            1. Repeat, while _next_ is not *false*,
+              1. Set _next_ to ? IteratorStep(_keysIter_).
+              1. If _next_ is not *false*, then
+                1. Let _nextValue_ be ? IteratorValue(_next_).
+                1. If _nextValue_ is *-0*<sub>𝔽</sub>, set _nextValue_ to *+0*<sub>𝔽</sub>.
+                1. NOTE: Because _other_ is an arbitrary object, it is possible for its *"keys"* iterator to produce the same value more than once.
+                1. Let _alreadyInResult_ be SetDataHas(_resultSetData_, _nextValue_).
+                1. Let _inThis_ be SetDataHas(_O_.[[SetData]], _nextValue_).
+                1. If _alreadyInResult_ is *false* and _inThis_ is *true*, then
+                  1. Append _nextValue_ to _resultSetData_.
+          1. Let _result_ be OrdinaryObjectCreate(%Set.prototype%, « [[SetData]] »).
+          1. Set _result_.[[SetData]] to _resultSetData_.
+          1. Return _result_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-set.prototype.isdisjointfrom">
+        <h1>Set.prototype.isDisjointFrom ( _other_ )</h1>
+        <p>This method performs the following steps when called:</p>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Perform ? RequireInternalSlot(_O_, [[SetData]]).
+          1. Let _otherRec_ be ? GetSetRecord(_other_).
+          1. If SetDataSize(_O_.[[SetData]]) ≤ _otherRec_.[[Size]], then
+            1. Let _thisSize_ be the number of elements in _O_.[[SetData]].
+            1. Let _index_ be 0.
+            1. Repeat, while _index_ &lt; _thisSize_,
+              1. Let _e_ be _O_.[[SetData]][_index_].
+              1. Set _index_ to _index_ + 1.
+              1. If _e_ is not ~empty~, then
+                1. Let _inOther_ be ToBoolean(? Call(_otherRec_.[[Has]], _otherRec_.[[Set]], « _e_ »)).
+                1. If _inOther_ is *true*, return *false*.
+                1. NOTE: The number of elements in _O_.[[SetData]] may have increased during execution of _otherRec_.[[Has]].
+                1. Set _thisSize_ to the number of elements in _O_.[[SetData]].
+          1. Else,
+            1. Let _keysIter_ be ? GetIteratorFromMethod(_otherRec_.[[Set]], _otherRec_.[[Keys]]).
+            1. Let _next_ be *true*.
+            1. Repeat, while _next_ is not *false*,
+              1. Set _next_ to ? IteratorStep(_keysIter_).
+              1. If _next_ is not *false*, then
+                1. Let _nextValue_ be ? IteratorValue(_next_).
```

Comment:
should this use IteratorStepValue? (and perhaps more places)

Reply:
Good catch, done.